File: 04-visitors/expressions/MulExpression.cpp
#include "MulExpression.h"

MulExpression::MulExpression(
    Expression *e1, Expression *e2
): first(e1), second(e2) {}

int MulExpression::eval() const {
    return first->eval() * second->eval();
}

void MulExpression::Accept(Visitor* visitor) {
    visitor->Visit(this);
}

File: 2023-fall/nodes/expressions/MulExpression.py
from nodes.expressions.Expression import Expression


class MulExpression(Expression):
    def __init__(
            self,
            left: Expression,
            right: Expression
        ) -> None:
        self.left = left
        self.right = right
    
    def accept(self, visitor):
        return visitor.visit_mul_expression(self)
